iron-collapse.html

```diff
@@ -128,6 +128,14 @@
       return this.horizontal ? 'width' : 'height';
     },
 
+    get dimensionMax() {
```

Comment:
Or even better, why not updating `dimension` to return `maxWidth, maxHeight`, and have a private utility function that converts from camel case to dashed?
```javascript
get dimension() {
  return this.horizontal ? 'maxWidth' : 'maxHeight';
},
/**
 * Convert a camel case string to dashed, e.g. maxWidth -> max-width.
 * @param {!String} string
 * @private 
 */
_toDash: function(string){
	return string.replace(/([A-Z])/g, function($1){return "-"+$1.toLowerCase();});
},

...

  this.style.transitionProperty = this._toDash(this.dimension);